File: 3D Multiplayer GP2/Assets/Modern Day Characters/Base Male/Scripts/JavaScript/GUI/ButtonSetAnimation.js
#pragma strict

@script RequireComponent(GUITexture);

public var animationChange : Animation;
public var animationName : String;
public var currentAnimation : String = "tPose";
public var matchNormalizedTime : boolean = true;

function Update() {
	if(Input.GetMouseButtonDown(0)) {
		if(GetComponent.<GUITexture>().HitTest(Input.mousePosition)) {
			animationChange.CrossFade(animationName);

			if(matchNormalizedTime) {
				animationChange[animationName].normalizedTime = animationChange[currentAnimation].normalizedTime;
			}

			transform.parent.BroadcastMessage("SetCurrentAnimation", animationName);
		}
	}
}

function SetCurrentAnimation(newAnimationName : String) {
	currentAnimation = newAnimationName;
}

File: 3D Multiplayer GP2/Assets/Modern Day Characters/Base Male/Scripts/C#/GUI/ButtonSetAnimation.cs
using System.Collections;
using System.Collections.Generic;
using UnityEngine;

[RequireComponent(typeof(GUITexture))]

public class ButtonSetAnimation : MonoBehaviour {
	public Animation animationChange;
	public string animationName;
	public string currentAnimation = "tPose";
	public bool matchNormalizedTime = true;
	
	void Update() {
		if(Input.GetMouseButtonDown(0)) {
			if(GetComponent<GUITexture>().HitTest(Input.mousePosition)) {
				animationChange.CrossFade(animationName);

				if(matchNormalizedTime) {
					animationChange[animationName].normalizedTime = animationChange[currentAnimation].normalizedTime;
				}

				transform.parent.BroadcastMessage("SetCurrentAnimation", animationName);
			}
		}
	}

	void SetCurrentAnimation(string newAnimationName) {
		currentAnimation = newAnimationName;
	}
}